extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.setup.js

```diff
@@ -54,15 +57,40 @@ angular.module('PaperUI.controllers.setup', []).controller('SetupPageController'
                 'enableChannels' : !$scope.advancedMode
             }, result.label).$promise.then(function() {
                 thingRepository.setDirty(true);
-                toastService.showDefaultToast('Thing added.', 'Show Thing', 'configuration/things/view/' + thingUID);
                 var thingType = thingTypeRepository.find(function(thingType) {
                     return thingTypeUID === thingType.UID;
                 });
 
-                if (thingType && thingType.bridge) {
-                    $scope.navigateTo('setup/search/' + thingUID.split(':')[0]);
-                } else {
-                    discoveryResultRepository.getAll(true);
+                var thing = thingRepository.find(function(thing) {
+                    return thing.UID === thingUID;
+                });
+
+                if (thing) {
+                    var configRequired = false;
+                    configDescriptionService.getByUri({
+                        uri : 'thing-type:' + thing.thingTypeUID
+                    }, function(configDescription) {
+                        if (configDescription.parameters) {
+                            for (var i = 0; i < configDescription.parameters.length; i++) {
+                                if (configDescription.parameters[i]['defaultValue'] === '' && configDescription.parameters[i]['required']) {
```

Comment:
In addition: You may also access the properties like `parameter.required`. Or is there any reason to use the associative array syntax?

Reply:
Thanks for pointing that out, it'll be done! But as a global question first: 
`required` and no `defaultValue` is a simple case, which should be handled here. 
But what is your opinion on `required` and empty `defaultValue===''`? 
Should an empty (but specified) defaultValue be handled here too or not? 
In `thing-type.xml` it would be: 
`            
<default></default>
<required>true</required>
`

Reply:
You are right, the empty default value can be ignored here. The empty `default` tag should not be accepted during review anyway.